sklearn/_display_estimator.py

```diff
@@ -0,0 +1,308 @@
+from sklearn._config import config_context
+from contextlib import closing
+from io import StringIO
+import uuid
+
+
+class _EstHTMLBlock:
+    """HTML Representation of Estimator
+
+    Parameters
+    ----------
+    type : {'serial', 'parallel', 'single'}
+        Type of HTML block
+
+    estimators : list of estimators or _EstHTMLBlock
+                or a single estimator
+        If type is in ('parallel', 'serial'), then `estimators` is a list of
+        estimators.
+        If type == 'single', then `estimators` is a single estimator.
+
+    names : list of str
+        If type in ('parallel', 'serial'), then `names` corresponds to
+        estimators
+        If type is 'single', then `names` is a single string corresponding to
+        the single estimator.
+
+    name_details : list of str, str, or None, default=None
+        If type == 'parallel', then `name_details` corresponds to `names`.
+        If type == 'single', then `name_details` is a single string
+        corresponding to the single estimator.
+        `name_details` is not used when type == 'single'.
+
+    dash_wrapped : bool, default=True
+        If true, wrapped HTML element will be wrapped with a dashed boarder.
+    """
+    def __init__(self, type, estimators, names, name_details=None,
+                 dash_wrapped=True):
+        self.type = type
+        self.estimators = estimators
+        self.names = names
+        self.dash_wrapped = dash_wrapped
+
+        if self.type == 'parallel' and name_details is None:
+            name_details = (None, ) * len(names)
+
+        self.name_details = name_details
+
+
+def _type_of_html_estimator(estimator):
```

Comment:
maybe this should be `_get_visual_block` or `_determine_block_type`... make it clearer in the calling context what the return type should be.

Reply:
actually, there's only one calling context... moving it closer to there might also help